cub/cub/device/dispatch/dispatch_select_if.cuh

```diff
@@ -407,23 +407,27 @@ __launch_bounds__(int(
  * @tparam OffsetT
  *   Signed integer type for global offsets
  *
- * @tparam SelectImpl SelectionOpt
+ * @tparam SelectionOpt
  *   SelectImpl indicating whether to partition, just selection or selection where the memory for the input and
  *   output may alias each other.
  */
-template <typename InputIteratorT,
-          typename FlagsInputIteratorT,
-          typename SelectedOutputIteratorT,
-          typename NumSelectedIteratorT,
-          typename SelectOpT,
-          typename EqualityOpT,
-          typename OffsetT,
-          SelectImpl SelectionOpt,
-          typename PolicyHub = detail::select::policy_hub<cub::detail::value_t<InputIteratorT>,
-                                                          cub::detail::value_t<FlagsInputIteratorT>,
-                                                          detail::select::per_partition_offset_t,
-                                                          (SelectionOpt == SelectImpl::SelectPotentiallyInPlace),
-                                                          (SelectionOpt == SelectImpl::Partition)>>
+template <
+  typename InputIteratorT,
+  typename FlagsInputIteratorT,
+  typename SelectedOutputIteratorT,
+  typename NumSelectedIteratorT,
+  typename SelectOpT,
+  typename EqualityOpT,
+  typename OffsetT,
+  SelectImpl SelectionOpt,
+  typename PolicyHub = detail::select::policy_hub<
+    detail::value_t<InputIteratorT>,
+    detail::value_t<FlagsInputIteratorT>,
+    // if/flagged/unique only have a single code path for different offset types, partition has different code paths
+    ::cuda::std::conditional_t<SelectionOpt == SelectImpl::Partition, OffsetT, detail::select::per_partition_offset_t>,
```

Comment:
I am passing the Offset type for partition here.

Reply:
Perfect, thanks!